lib/Gedmo/Translatable/TranslatableListener.php

```diff
@@ -666,7 +666,11 @@ private function handleTranslatableObjectUpdate(TranslatableAdapter $ea, $object
                 $translation->setContent($content);
                 // check if need to update in database
                 $transWrapper = AbstractWrapper::wrap($translation, $om);
-                if (((is_null($content) && !$isInsert) || is_bool($content) || is_int($content) || (is_string($content) && strlen($content) > 0) || !empty($content)) && ($isInsert || !$transWrapper->getIdentifier() || isset($changeSet[$field]))) {
+                if ($content === null && !$isInsert) {
+                    // if the $content is null'ed remove the full translation to avoid fragments
+                    $om->clear($translation);
```

Comment:
Calling `clear` here won't work. First of all (at least for ODM) it takes a document name, not a document itself. Second of all, when the document has been cleared from the object manager, the call to `remove` won't do a thing since the document in question is not managed.